azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java

```diff
@@ -0,0 +1,51 @@
+/*
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT License. See License.txt in the project root for
+  license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.model;
+
+import lombok.Getter;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+
+public interface IArtifact {
+    String getId();
```

Comment:
why not a super POJO ? I feel that "id" is only a property of a object.

Reply:
artifact is not a azure resource like `mysql` or `webapp`/`function`. 
and I dont see the necessity to abstract them at least for now and we may do this abstraction in the future when needed.